src/clr-angular/data/datagrid/datagrid.spec.ts

```diff
@@ -300,17 +301,47 @@ export default function(): void {
 
       describe('Template API', function() {
         it('sets the currentSingle binding', function() {
+          expect(selection.currentSingle).toBeUndefined();
+          context.testComponent.selected = 1;
+          context.detectChanges();
+          expect(selection.currentSingle).toEqual(1);
+          context.testComponent.selected = null;
+          context.detectChanges();
           expect(selection.currentSingle).toBeNull();
+        });
+
+        it('does not emit a change event for initial value of currentSingle that is undefined', function() {
```

Comment:
I think we want to break this up into separate tests.


1. `it does not emit a change event on initialization, before selection`
2. `it does not emit a change event when selecting the same value`
3. `it does not emit a change event when setting selection to undefined if already null, and vice versa`
4. `it emits a change event when changing the selection`

Reply:
Will break it up into four tests.